server/src/main/java/org/opensearch/action/admin/cluster/shards/routing/wrr/get/TransportGetWRRWeightsAction.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.action.admin.cluster.shards.routing.wrr.get;
+
+import org.opensearch.action.ActionListener;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import org.opensearch.action.ActionRequestValidationException;
+import org.opensearch.action.support.ActionFilters;
+import org.opensearch.action.support.clustermanager.TransportClusterManagerNodeReadAction;
+import org.opensearch.cluster.ClusterState;
+import org.opensearch.cluster.block.ClusterBlockException;
+import org.opensearch.cluster.metadata.IndexNameExpressionResolver;
+import org.opensearch.cluster.metadata.Metadata;
+
+import org.opensearch.cluster.metadata.WeightedRoundRobinRoutingMetadata;
+import org.opensearch.cluster.node.DiscoveryNode;
+import org.opensearch.cluster.routing.WRRWeights;
+import org.opensearch.cluster.routing.allocation.decider.AwarenessAllocationDecider;
+import org.opensearch.cluster.service.ClusterService;
+import org.opensearch.common.inject.Inject;
+import org.opensearch.common.io.stream.StreamInput;
+
+import org.opensearch.common.settings.ClusterSettings;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.threadpool.ThreadPool;
+import org.opensearch.transport.TransportService;
+
+import java.io.IOException;
+import java.util.List;
+
+import static org.opensearch.action.ValidateActions.addValidationError;
+
+/**
+ * Transport action for getting weights for weighted round-robin search routing policy
+ *
+ * @opensearch.internal
+ */
+public class TransportGetWRRWeightsAction extends TransportClusterManagerNodeReadAction<
```

Comment:
should we extend `HandledTransportAction` instead of `TransportClusterManagerNodeReadAction` ? I do see that `TransportClusterManagerNodeReadAction` are used for GET actions which needs cluster state . but just thinking about the benefits/cost for the same .